### ISSUE TITLE
Feature request: Add ReplicationController and Daemonsets metrics 

### ISSUE THREAD

**Issue by natalia-k** [kubernetes/kube-state-metrics#50]
Hello,
Is it possible to add ReplicationController and Daemonsets metrics  as well?
I am using the following metrics to alert  if number of running pods is not equal to replicas define in rc, ds : 
    rc_status_readyReplicas, rc_spec_replicas 
   ds_status_currentNumberScheduled, ds_status_desiredNumberScheduled

I find also, that Kubernetes API don't provide the rc_status_readyReplicas metric at all when it's 0 - how this can be fixed ?
Thanks!

**Comment by brancz** [comment]
> For context: this issue was opened due to a [discussion](https://groups.google.com/forum/#!topic/prometheus-developers/4GAHi_qrC1o) on the Prometheus Developers mailing list.

Thanks for opening this issue @paytinka .

> I find also, that Kubernetes API don't provide the rc_status_readyReplicas metric at all when it's 0 - how this can be fixed ?

This behavior is simply because the exporter you are using is written in Python and it just parses the json/yaml to a python dict and therefore when a field is missing it does not end up in the dict (please correct me if I'm wrong @tomwilkie, as I'm not very familiar with the code). This exporter unmarshals the objects into the structs, therefore when a field is left empty in the json/yaml then its type's default value is chosen - which is why the readyReplicas field will end up being 0 and the metric is exposed correctly.

I currently don't have much bandwidth to work on this, but if anyone has time I'm happy to review it.

**Comment by paytinka** [comment]
here is example, I changed port of readinessProbe to wrong port :
```
%kubectl --namespace=kube-system get rc
NAME                          DESIRED   CURRENT   READY     AGE
kube-dns-v20                     3         3        0       41d
```

`%kubectl get rc kube-dns-v20 -o json --namespace=kube-system`
...
    "status": {
        "replicas": 3,
        "fullyLabeledReplicas": 3,
        "observedGeneration": 11
    }

When I change it back to the correct port: 
`%kubectl get rc kube-dns-v20 -o json --namespace=kube-system`
...
    "status": {
        "replicas": 3,
        "fullyLabeledReplicas": 3,
        "**readyReplicas**": 3,
        "observedGeneration": 12
    }
As you can see, in case of READY = 0 , the metrics readyReplicas not appeared in json output at all.

Thanks
Natalia

**Comment by brancz** [comment]
Yes that makes sense, however, due to how the unmarshaling is done with go and the libraries we use in this project, it is no issue. We simply need to add support for `ReplicationController`s and `DaemonSet`s.

What would happen in `kube-state-metrics` is that the json/yaml is unmarshaled into [this](https://github.com/kubernetes/client-go/blob/843f7c4f28b1f647f664f883697107d5c02c5acc/1.5/pkg/api/v1/types.go#L2042) struct. Kubernetes simply skips the `ReadyReplicas` field, if unset, when marshaling it into the respective representation (due to the `omitempty` annotation). That results in: since we unmarshal it into that exact same struct we know that it should be there and if it's not we know it's 0.

**Comment by nleib** [comment]
+1

**Comment by dominikschulz** [comment]
See #56 and #54.

**Comment by brancz** [comment]
Closed by #56 and #54
